### PR TITLE
[sdk-metrics] Include Meter.Tags in metric identity resolution

### LINKED ISSUE #5859: [bug] Mark instrumentation scope attributes as identifying
### Package

OpenTelemetry

### Package Version

| Package Name                          | Version |
| ------------------------------------- | ------- |
| OpenTelemetry.Api                     | 1.8.0   |
| OpenTelemetry                         | 1.8.0   |
| TBD                                   | TBD     |

### Runtime Version

N/A

### Description

Recently spec made scope tags as identifying. Only Metrics are affected today in this repo, but when we add Scope tags for Activitysource for .NET9, this should be considered.
https://github.com/open-telemetry/opentelemetry-specification/pull/4161#issuecomment-2261242491

### Steps to Reproduce

https://github.com/open-telemetry/opentelemetry-specification/pull/4161#issuecomment-2261242491 shows the unittest

### Expected Result

make scope properties identifying

### Actual Result

https://github.com/open-telemetry/opentelemetry-specification/pull/4161#issuecomment-2261242491

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by dulikvor**
Fixes #5859 
Design discussion issue #

## Changes

Metric tags are now considered identifiers, contributing to metric scope uniqueness.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [x] Unit tests added/updated
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: src/OpenTelemetry/Metrics/Tags.cs [hunk 1]
```diff
@@ -27,6 +27,39 @@ public Tags(KeyValuePair<string, object?>[] keyValuePairs)
 
     public static bool operator !=(Tags tag1, Tags tag2) => !tag1.Equals(tag2);
 
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static int ComputeHashCode(KeyValuePair<string, object?>[] keyValuePairs)
```
Comment (cijothomas) [inline src/OpenTelemetry/Metrics/Tags.cs:31]: Don't think this change is required.

**Review by cijothomas — APPROVED** [review]
Thanks!
Please add changelog entry too, as this is breaking existing behavior to fix bug.

File: src/OpenTelemetry/Metrics/Tags.cs (hunk 1 above)
Comment (dulikvor) [reply src/OpenTelemetry/Metrics/Tags.cs]: Made sure the test is not dependent and reverted the change.

File: src/OpenTelemetry/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,10 @@ Notes](../../RELEASENOTES.md).
 
 ## Unreleased
 
+* Updated `OpenTelemetry.Metrics.MetricStreamIdentity` Metric tags are now considered identifiers,
```
Comment (rajkumar-rangaraj) [inline src/OpenTelemetry/CHANGELOG.md:9]:
```suggestion
* Update Metric tags to be identifying for the Meter, aligning with the specification. Previously, Metric tags were omitted from being identifying.
```

**Review by rajkumar-rangaraj — APPROVED** [review]

File: src/OpenTelemetry/CHANGELOG.md (hunk 2 above)
Comment (CodeBlanch) [reply src/OpenTelemetry/CHANGELOG.md:9]: Will anyone reading this understand it? 🤣

How about...

* [Meter.Tags](https://learn.microsoft.com/dotnet/api/system.diagnostics.metrics.meter.tags) will now be considered when resolving the SDK metric to update when measurements are recorded. Meters with the same name and different tags will now lead to unique metrics.

PS: There is a lint error can't merge until that is resolved.

Comment (dulikvor) [reply src/OpenTelemetry/CHANGELOG.md]: Fine with the latter, made the change. can linter re instigate, I will check if its yet too long.

**Review by Kielek — APPROVED** [review]

**Review by CodeBlanch — APPROVED** [review]
LGTM

**Pull request merged**